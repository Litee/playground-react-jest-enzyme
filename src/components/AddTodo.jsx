import React, { Component } from 'react'

class AddTodo extends Component {
  constructor (props) {
    super(props)
    this.state = { value: '' }
    this.handleChange = e => {
      this.setState({ value: e.target.value })
    }
    this.handleSubmit = e => {
      this.props.onAddTodo(this.state.value)
      this.setState({ value: '' })
      e.preventDefault()
    }
  }

  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          type='text'
          value={this.state.value}
          onChange={this.handleChange}
        />
        <button onClick={this.handleSubmit}>add</button>
      </form>
    )
  }
}

export default AddTodo
