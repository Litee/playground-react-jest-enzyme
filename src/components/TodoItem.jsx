import React, { Component } from 'react'

class TodoItem extends Component {
  render () {
    return (
      <li>
        {this.props.title} <button onClick={this.props.onDelete}>delete</button>
      </li>
    )
  }
}

export default TodoItem
