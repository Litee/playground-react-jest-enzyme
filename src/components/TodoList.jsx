import React from 'react'
import TodoItem from './TodoItem'
import AddTodo from './AddTodo'

export default class TodoList extends React.Component {
  constructor () {
    super()
    this.state = {
      todos: []
    }
    this.onAddTodo = newTodo => {
      this.setState(prevState => {
        return {
          todos: prevState.todos.some(t => t === newTodo)
            ? prevState.todos
            : [...prevState.todos, newTodo]
        }
      })
    }
    this.onDeleteTodo = todoToRemove => {
      this.setState(prevState => {
        return {
          todos: prevState.todos.filter(t => t !== todoToRemove)
        }
      })
    }
  }

  render () {
    const items = [...this.state.todos].map(todo => (
      <TodoItem
        key={todo}
        title={todo}
        onDelete={this.onDeleteTodo.bind(this, todo)}
      />
    ))
    return (
      <div>
        <h2>TODO Items:</h2>
        <ul>{items}</ul>
        <AddTodo onAddTodo={this.onAddTodo} />
      </div>
    )
  }
}
