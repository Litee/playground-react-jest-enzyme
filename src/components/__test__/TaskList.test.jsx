import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import TodoList from '../TodoList'
import TodoItem from '../TodoItem'
import AddTodo from '../AddTodo'

function addTodo (wrapper, title) {
  wrapper.find(AddTodo).find('input').simulate('change', { target: {value: title} })
  wrapper.find(AddTodo).find('form').simulate('submit', {preventDefault: () => {}})
}

test('Must render properly', () => {
  const wrapper = shallow(<TodoList />)
  expect(shallowToJson(wrapper)).toMatchSnapshot()
})

test('Must add todo', () => {
  const wrapper = mount(<TodoList />)
  addTodo(wrapper, 'Do X')
  expect(shallowToJson(wrapper)).toMatchSnapshot()
})

test('Must ignore adding duplicates', () => {
  const wrapper = mount(<TodoList />)

  addTodo(wrapper, 'Do X')
  addTodo(wrapper, 'Do X')
  expect(shallowToJson(wrapper)).toMatchSnapshot()
})

test('Must remove todo', () => {
  const wrapper = mount(<TodoList />)

  addTodo(wrapper, 'Do X')
  wrapper.find(TodoItem).find('button').simulate('click')
  expect(shallowToJson(wrapper)).toMatchSnapshot()
})
