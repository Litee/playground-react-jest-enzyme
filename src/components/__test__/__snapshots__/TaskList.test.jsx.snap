// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must add todo 1`] = `
<TodoList>
  <div>
    <h2>
      TODO Items:
    </h2>
    <ul>
      <TodoItem
        key="Do X"
        onDelete={[Function]}
        title="Do X"
      >
        <li>
          Do X
           
          <button
            onClick={[Function]}
          >
            delete
          </button>
        </li>
      </TodoItem>
    </ul>
    <AddTodo
      onAddTodo={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
        >
          add
        </button>
      </form>
    </AddTodo>
  </div>
</TodoList>
`;

exports[`Must ignore adding duplicates 1`] = `
<TodoList>
  <div>
    <h2>
      TODO Items:
    </h2>
    <ul>
      <TodoItem
        key="Do X"
        onDelete={[Function]}
        title="Do X"
      >
        <li>
          Do X
           
          <button
            onClick={[Function]}
          >
            delete
          </button>
        </li>
      </TodoItem>
    </ul>
    <AddTodo
      onAddTodo={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
        >
          add
        </button>
      </form>
    </AddTodo>
  </div>
</TodoList>
`;

exports[`Must remove todo 1`] = `
<TodoList>
  <div>
    <h2>
      TODO Items:
    </h2>
    <ul />
    <AddTodo
      onAddTodo={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
        >
          add
        </button>
      </form>
    </AddTodo>
  </div>
</TodoList>
`;

exports[`Must render properly 1`] = `
<div>
  <h2>
    TODO Items:
  </h2>
  <ul />
  <AddTodo
    onAddTodo={[Function]}
  />
</div>
`;
